src/ring/adapter/jetty9.clj

```diff
@@ -250,20 +255,36 @@
                                           job-queue)
                    (.setDaemon daemon?)))
         server (doto (Server. ^ThreadPool pool)
-                 (.addBean (ScheduledExecutorScheduler.)))
+                 (.addBean (ScheduledExecutorScheduler.))
+                 (.addBean (proxy [AbstractLifeCycle] []
+                             (doStart [] (when-some [f (:lifecycle-start options)] (f)))
+                             (doStop  [] (when-some [f (:lifecycle-end options)] (f)))))
```

Comment:
nit: is there any context we can provide to the `f`? for example, the server.

Reply:
As I understand it, this mechanism (`.addBean`) exists so that external objects can essentially inherit the lifecycle of the server. For example, a typical implementation of the `.doStop()` method would be to do some logging and shutdown the external object. Therefore, I don't see any obvious arguments to pass here.